doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -473,82 +473,112 @@ Induction
 
-.. tacv:: discriminate
+.. tacn:: injection {? @induction_arg } {? as {* @simple_intropattern } }
 
-   This behaves like :n:`discriminate @ident` if ident is the name of an
-   hypothesis to which ``discriminate`` is applicable; if the current goal is of
-   the form :n:`@term <> @term`, this behaves as
-   :n:`intro @ident; discriminate @ident`.
+   Exploits the property that constructors of
+   inductive types are injective, i.e. that if :n:`c` is a constructor of an
+   inductive type and :n:`c t__1` and :n:`c t__2` are equal then
+   :n:`t__1` and :n:`t__2` are equal too.
 
-   .. exn:: No discriminable equalities.
-      :undocumented:
+   If there is a hypothesis `H` in the form :n:`@term__1 = @term__2`,
+   then :n:`injection H` applies the injectivity of constructors as deep as
+   possible to derive the equality of subterms of :n:`@term__1` and
+   :n:`@term__2` wherever the subterms start to differ. For example, from
+   :g:`(S p, S n) = (q, S (S m))` we may derive :g:`S p = q` and
+   :g:`n = S m`. The terms must have inductive type and the same head
+   constructor, but must not be convertible. If so, the tactic derives the
+   equalities and adds them to the current goal as :term:`premises <premise>`.
 
-.. tacn:: injection @term
-   :name: injection
+   If the current goal is of the form :n:`@term <> @term` and no :n:`induction_arg`
+   is provided, :tacn:`injection` is equivalent to :n:`intro @ident; injection @ident`.
 
-   The injection tactic exploits the property that constructors of
-   inductive types are injective, i.e. that if :g:`c` is a constructor of an
-   inductive type and :g:`c t`:sub:`1` and :g:`c t`:sub:`2` are equal then
-   :g:`t`:sub:`1` and :g:`t`:sub:`2` are equal too.
+   :n:`@natural` (in :n:`induction_arg`)
+     Equivalent to :tacn:`intros` :n:`until @natural` followed by
+     :n:`injection @ident` where :n:`@ident` is the identifier for the last
+     introduced hypothesis.
 
-   If :n:`@term` is a proof of a statement of conclusion :n:`@term = @term`,
-   then :tacn:`injection` applies the injectivity of constructors as deep as
-   possible to derive the equality of all the subterms of :n:`@term` and
-   :n:`@term` at positions where the terms start to differ. For example, from
-   :g:`(S p, S n) = (q, S (S m))` we may derive :g:`S p = q` and
-   :g:`n = S m`. For this tactic to work, the terms should be typed with an
-   inductive type and they should be neither convertible, nor having a different
-   head constructor. If these conditions are satisfied, the tactic derives the
-   equality of all the subterms at positions where they differ and adds them as
-   antecedents to the conclusion of the current goal.
+   :n:`@one_term with @bindings`  (in :n:`@induction_arg`)
+      Like :n:`injection @term` but uses the given bindings to
+      instantiate parameters or hypotheses of :n:`@one_term`.
+
+   .. todo PR: what does "injection @ident" do?  Just tells Coq which
+      hypothesis to use?
```

Comment:
That's correct. Note that this is a subcase of `injection @one_term`, where `@one_term` is supposed to be the proof of a primitive equality.

Reply:
"primitive equality" corresponds to Leibniz, setoid or definitional equality?  Better to use one of those terms.

Reply:
It corresponds to Leibniz = definitional equality. I agree it would be better to use one of these two defined terms.

Reply:
Huh?  Leibniz and definitional equality are different things.  You mean just Leibniz, right?

Reply:
Yes, I just mean Leibniz, I was confused while writing this review (thinking of too many other things at the same time).